### PR TITLE
Updated generated Sql.Sdk

### PR DESCRIPTION
**Pull request by vidai-msft**
## Description

Re-generated Sql.Sdk using AutoRest and updated all the modified files.

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Sql/Sql.Sdk/Generated/Models/ExtendedServerBlobAuditingPolicy.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ public ExtendedServerBlobAuditingPolicy()
         /// audit logs.
         ///
         /// For more information, see [Database-Level Audit Action
-        /// Groups](https://learn.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-action-groups).
```
Comment (milanbrkic-ms) [inline src/Sql/Sql.Sdk/Generated/Models/ExtendedServerBlobAuditingPolicy.cs:119]: Shouldn't we keep `learn.microsoft` links?

File: src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs [hunk 2]
```diff
@@ -94,6 +94,21 @@ public async Task<AzureOperationResponse<BackupShortTermRetentionPolicy>> GetWit
...
+            {
+                if (serverName.Length > 63)
+                {
+                    throw new ValidationException(ValidationRules.MaxLength, "serverName", 63);
+                }
+                if (serverName.Length < 3)
+                {
+                    throw new ValidationException(ValidationRules.MinLength, "serverName", 3);
+                }
+                if (!System.Text.RegularExpressions.Regex.IsMatch(serverName, "^[a-z][a-z0-9]*$"))
```
Comment (milanbrkic-ms) [inline src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs:107]: server name can accept hyphens as well (`-`)

Comment (ostojicb) [reply src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs:107]: @vidai-msft where do these validations originate from?

File: src/Sql/Sql.Sdk/Generated/Models/ExtendedServerBlobAuditingPolicy.cs (hunk 1 above)
Comment (vidai-msft) [reply src/Sql/Sql.Sdk/Generated/Models/ExtendedServerBlobAuditingPolicy.cs:119]: Autorest.csharp has not updated this static string I believe as we are using an old version for track1.

File: src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs [hunk 3]
```diff
@@ -94,6 +94,21 @@ public async Task<AzureOperationResponse<BackupShortTermRetentionPolicy>> GetWit
             {
                 throw new ValidationException(ValidationRules.CannotBeNull, "serverName");
             }
+            if (serverName != null)
```
[the remark this answers is not in the document]
Comment (vidai-msft) [reply src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs:97]: Looks like the commit id involved new features. Previously, we don't see such kind of validation.

**Comment by vidai-msft** [comment]
@milanbrkic-ms Is it OK to use this commit id for your PR c9fd3070a45da18959192265c0e62dd34b521ab1? I have confirmed the breaking change was introduced from the previous commit id and in this commit, the pattern validation was reverted.

File: src/Sql/Sql.Sdk/Generated/Models/ExtendedServerBlobAuditingPolicy.cs (hunk 1 above)
Comment (vidai-msft) [reply src/Sql/Sql.Sdk/Generated/Models/ExtendedServerBlobAuditingPolicy.cs]: Reverted the change

File: src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs (hunk 2 above)
Comment (vidai-msft) [reply src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs]: Used a new commit id where the serverName validation was reverted.

File: src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (vidai-msft) [reply src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs]: Confirmed the breaking change was introduced with the old commit id we used. Now I replaced it with a new commit id.

File: src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs (hunk 2 above)
Comment (vidai-msft) [reply src/Sql/Sql.Sdk/Generated/BackupShortTermRetentionPoliciesOperations.cs]: serverName validation was reverted.

**Review by milanbrkic-ms — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Comment by vidai-msft** [comment]
@milanbrkic-ms we have merged the code. Can you please fetch the latest and then re-generate the Sql.Sdk in your PR to see if anything would change? Thanks!

**Pull request merged**